03_letters/srini/main_test.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := `a:2
+b:1`
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestLetters(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	letterMap := letters("abba")
+
+	// Then
+	expected := map[rune]int{rune('a'): 2, rune('b'): 2}
+
+	r.Equalf(expected, letterMap, "Unexpected output in main()")
+}
+
+func TestLettersWithSpaces(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	letterMap := letters("abb a")
```

Comment:
NICE!
Last comment seeing as you are so ahead already: consider using [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests). Include empty string.

Reply:
Also, please `Update branch`